### PR TITLE
fix(environments): Support environments provided within quotation marks

### PR DESCRIPTION
**Pull request by lynnagara**

### CODE REVIEW

File: src/sentry/static/sentry/app/utils/queryString.jsx [hunk 1]
```diff
@@ -4,10 +4,20 @@ export function formatQueryString(qs) {
...
 
 // returns environment name from query or null if not specified
-// Any charater can be valid in an environment name
+// Any character can be valid in an environment name but we need to
+// check for matching environments with the quotation marks first
+// to match the way tag searches are being done
 export function getQueryEnvironment(qs) {
-  const match = qs.match(/environment:([^\s]*)/);
-  return match ? match[1] : null;
+  const matchWithQuotes = qs.match(/environment:"([^\s]*)"/);
```
Comment (billyvg) [inline src/sentry/static/sentry/app/utils/queryString.jsx]: I was going to say that you can do it in 1 regex e.g. `/environment:("|)([^\s]*)\1/`, but we probably want to allow spaces when quotes are used. I'm sure there's a way to do this in 1 regex (cc self-proclaimed regex god @benvinegar) but having 2 will allow us lesser mortals to understand it better.

Also, if we allow quotes to be used, then how do we allow searching for literal quotes, etc. We should start defining what we want to allow in our test cases.

**Review by billyvg — CHANGES_REQUESTED** [review]

File: src/sentry/static/sentry/app/utils/queryString.jsx [hunk 2]
```diff
@@ -4,10 +4,20 @@ export function formatQueryString(qs) {
 }
 
 // returns environment name from query or null if not specified
-// Any charater can be valid in an environment name
+// Any character can be valid in an environment name but we need to
```
[the remark this answers is not in the document]
Comment (billyvg) [reply src/sentry/static/sentry/app/utils/queryString.jsx]: err see below comment

File: src/sentry/static/sentry/app/utils/queryString.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/utils/queryString.jsx]: Maybe allow a set of delimiters?

```
'"`|/\
```

Comment (lynnagara) [reply src/sentry/static/sentry/app/utils/queryString.jsx]: Updated to support spaces

**Review by billyvg — APPROVED** [review]

**Pull request merged**